test/lib/visualTestLoop.js

```diff
@@ -175,8 +200,13 @@
               testName + ' [' + golden + '] has too many different pixels ' + differentPixels + '(' + okDiff + ') representing ' + percDiff + '% (>' + (percentage * 100) + '%)'
             );
             if (!isOK) {
-              var stringa = imageDataToChalk(output);
-              console.log(stringa);
+              // var stringa = imageDataToChalk(output);
+              // console.log(stringa);
+              try {
+                dumpFailedTest(testName, renderedCanvas, canvas, output);
```

Comment:
you can move this into the if block below

Reply:
i prefer it to happen every run, in case i can link the results out in PRs

Reply:
I looked into that as well
But you said that node visual test results are not accurate.
That is why I disabled the option the debug visuals in node tests to begin with and moved to browser outputs.

Reply:
I can be done if we post the results somewhere.
Github doesn't allow creating a comment with images using actions/bots but we can easily add links to posted images.
Maybe we create a test dumps repo under `fabricjs` org and commit to there

Reply:
we create a branch, push images and remove the branch once the pr is closed

Reply:
BTW you should use the method `generateGolden` by changing the path output instead of adding more logic IMO

Reply:
How about the following:
CI runs visual tests with `-d` flag, meaning that failing tests update golden refs.
Then it creates a branch with these changes and posts a comment with the diff between the branches.
It can prefix the branch name with something and delete it once the pr is closed.
Thoughts?

Reply:
#8210

Reply:
I didn't add a generate golden method, the method is differnet and produces a collage of the 3 images.
This method replaced the imageDataToChalk function that was not really usefull.
I want to take care of this, i felt the need for it, i started it, i can finish it. Then if is really ugly we can improve it.


> we create a branch, push images and remove the branch once the pr is closed

This seems an overkill of resourcers.

There are actions that archive artifacts of pr runs, and keep them somewhere in github.

Reply:
I looked at artifacts...
Am not sure how to display the results though
This on the other hand is simple: https://github.com/fabricjs/fabric.js/compare/ci-visual-test-action...ci-pr%238210-visuals-firefox

Reply:
Maybe is an overkill, but it is on github actions...
This is the result:
https://github.com/fabricjs/fabric.js/pull/8210#issuecomment-1229892565